### ISSUE TITLE
Error:  Something went wrong installing the "sharp" module

### ISSUE THREAD

**Issue by david-gagne** [gridsome/gridsome#1503]
## Description

Running the `gridsome develop` command results in an unexpected error related to the "sharp" module. 

### Steps to reproduce

1. Copy the "Tulip" starter template: `gridsome create my-project Uninen/gridsome-starter-tulip` (__Note:__ This issue seems to be present using any starter template).
2. Run `cd my-project` as a workaround for _"Error: Failed to install dependencies with yarn. Please enter my-project directory and install dependencies with yarn or npm manually. Then run gridsome develop to start local development."_
3. Run `gridsome develop` in my-project directory.

### Expected result

Expected a local development server at http://localhost:8080

### Actual result

Received the following error: `Error: 
Something went wrong installing the "sharp" module`

### Environment

System:
    OS: macOS 11.2.3
    CPU: (8) arm64 Apple M1
  Binaries:
    Node: 16.1.0 - ~/.nvm/versions/node/v16.1.0/bin/node
    Yarn: 1.22.10 - /opt/homebrew/bin/yarn
    npm: 7.11.2 - ~/.nvm/versions/node/v16.1.0/bin/npm
  Browsers:
    Chrome: 90.0.4430.212
    Safari: 14.0.3
  npmPackages:
    @gridsome/plugin-google-analytics: ^0.1.2 => 0.1.2 
    gridsome: ^0.7.23 => 0.7.23 
    gridsome-plugin-tailwindcss: ^3.0.1 => 3.0.1 
  npmGlobalPackages:
    @gridsome/cli: 0.3.4

**Comment by egonbraun** [comment]
This happens due to `sharp` package needing `libvips` M1 binary available which is not by default on MacOS, therefore it tries to build it and then it fails to do so.

While we still do not have `libvips` binaries available for Mac M1 machines you will have to go around this issue by installing `libvips` using `homebrew`.

```bash
$ brew install libvips
```

Please, note that the **con** on this process is that `libvips` has lots of dependencies.

Hope it helps!

**Comment by david-gagne** [comment]
@egonbraun I installed `libvips` as you recommended, but I'm unsure how to proceed from here because installing `libvips` didn't fix the issue with the `sharp` package. Do you have any further recommendations on how I should proceed?

**Comment by egonbraun** [comment]
> @egonbraun I installed `libvips` as you recommended, but I'm unsure how to proceed from here because installing `libvips` didn't fix the issue with the `sharp` package. Do you have any further recommendations on how I should proceed?

In my case, I first installed `libvips` and then reinstalled `gridsome` using `npm`. For me, I had the issue when installing gridsome and this fixed it for me. Maybe you should look into your environment variables, check:

```bash
$ brew shellenv libvips
```

**Comment by david-gagne** [comment]
@egonbraun it worked! 
<img width="456" alt="Screen Shot 2021-05-26 at 23 49 38" src="https://user-images.githubusercontent.com/12618094/119763185-11937d00-be7d-11eb-9971-e637ea42c4d5.png">

You rock! Thanks for holding my hand through this!

**Comment by mattaningram** [comment]
@egonbraun I'm not sure this is the case anymore, according to this `sharp` issue it can successfully be built on M1 Macs now. Any reason Gridsome hasn't updated sharp to use this version? https://github.com/lovell/sharp/issues/2460

**Comment by weaversam8** [comment]
> > @egonbraun I installed `libvips` as you recommended, but I'm unsure how to proceed from here because installing `libvips` didn't fix the issue with the `sharp` package. Do you have any further recommendations on how I should proceed?
> 
> In my case, I first installed `libvips` and then reinstalled `gridsome` using `npm`. For me, I had the issue when installing gridsome and this fixed it for me. Maybe you should look into your environment variables, check:
> 
> ```shell
> $ brew shellenv libvips
> ```

Adding a note here on successfully executing this workaround:

In case it isn't obvious, you may need to eval the environment variables brew outputs (execute `eval $(brew shellenv libvips)`), and I also needed to `brew link --overwrite` both `imath` and `openexr` (and you may need to install those packages if they weren't already.)

**Comment by humblecoder** [comment]
I've completed each recommended step in this issue, and have taken a look at the linked issue -- still receiving the same `sharp` error.  What's the actual problem here?  Does the team need to update the sharp version, or does a replacement (assuming one exists) need to be found?  

Are there any general updates otherwise?  🤔

**Comment by alexmarple** [comment]
I ran into this same problem and definitely tried all of the above and was coming up with nothing. 

What finally fixed it? 

1. Well first I finally listened to my computer and upgrade to to macOS 12.4. Then I was able to get the Gridsome CLI to properly download. 
2. I was still running into the sharp issue so I found someone who put this bit of code in their package.json.
```
  "resolutions": {
    "gridsome/sharp": "0.29.3"
  }
```
After that it finally all worked no problem. Your mileage may vary but I wanted to put this here for anyone who happens upon this thread.
